### PR TITLE
Remove doc comments from concrete floating-point types

### PR DESCRIPTION
**Pull request by stephentyrone**
... when they are already present on the protocol. I don't *think* that we need them anymore for xcode documentation purposes. There are reasonable arguments both ways on this:
1. when you're editing the concrete implementations, it's sometimes nice to have the doc comment right there.
2. but it needlessly repetitive, and introduces the opportunity for comments to get out of sync.
3. it also adds noise; it would be nice for information density if the implementation only had implementation notes.

Also includes some minor formatting fixups, and a bugfix for isCanonical on platforms that do not support subnormals (armv7, currently, which is sufficiently minor that I'm OK with stealth fixing it like this).

### CODE REVIEW

**Comment by stephentyrone** [comment]
@swift-ci please test

File: stdlib/public/core/FloatingPointTypes.swift.gyb [hunk 1]
```diff
@@ -231,89 +196,60 @@ extension ${Self}: BinaryFloatingPoint {
...
-  /// type exist, and noncanonical values may exist for other types that
-  /// conform to the `FloatingPoint` protocol.
-  ///
-  /// [spec]: http://ieeexplore.ieee.org/servlet/opac?punumber=4610933
   @inlinable
   public var isCanonical: Bool {
+    // All Float and Double encodings are canonical in IEEE 754.
+    //
+    // On platforms that do not support subnormals, we treat them as
+    // non-canonical encodings of zero.
```
Comment (xwu) [inline stdlib/public/core/FloatingPointTypes.swift.gyb]: I think we should document this in the corresponding protocol requirement in `FloatingPoint.swift`. I like LLVM's description that offers examples:

> Examples of non-canonical encodings:
>
> * x87 pseudo denormals, pseudo NaNs, pseudo Infinity, Unnormals. These are converted to a canonical representation per hardware-specific protocol.
> * Many normal decimal floating-point numbers have non-canonical alternative encodings.
> * Some machines, like GPUs or ARMv7 NEON, do not support subnormal values. These are treated as non-canonical encodings of zero[...]

File: stdlib/public/core/FloatingPointTypes.swift.gyb [hunk 2]
```diff
@@ -883,53 +600,30 @@ extension ${Self}: BinaryFloatingPoint {
...
         exponentBitPattern: UInt(Int(${Self}._exponentBias) + clamped),
-        significandBitPattern: 0)
+        significandBitPattern: 0
+      )
       result = result * scale
     }
     self = result
   }
 
-  /// Creates a NaN ("not a number") value with the specified payload.
```
Comment (xwu) [inline stdlib/public/core/FloatingPointTypes.swift.gyb]: This isn't a protocol requirement, so we can't delete the documentation here.

File: stdlib/public/core/FloatingPointTypes.swift.gyb (hunk 1 above)
Comment (stephentyrone) [reply stdlib/public/core/FloatingPointTypes.swift.gyb]: Done!

File: stdlib/public/core/FloatingPointTypes.swift.gyb (hunk 2 above)
Comment (stephentyrone) [reply stdlib/public/core/FloatingPointTypes.swift.gyb]: Restored.

**Comment by stephentyrone** [comment]
@swift-ci please smoke test

**Comment by stephentyrone** [comment]
@natecook1000 is on vacation this week, I'm going to wait to merge this until he has a chance to give it at least a cursory approval.

**Review by natecook1000 — COMMENTED** [review]
Looks great, with some minor nits! I’m especially glad to see the repeated docs dropping off the extension methods. Do we have tests that show that the docs are being correctly inherited from the protocols down onto the concrete symbols?

**Comment by stephentyrone** [comment]
> Do we have tests that show that the docs are being correctly inherited from the protocols down onto the concrete symbols?

I don't think this is testable for Swift, since it's an Xcode thing, but it seems to work robustly in 11.

File: stdlib/public/core/FloatingPoint.swift [hunk 3]
```diff
@@ -1197,13 +1197,27 @@ public protocol FloatingPoint: SignedNumeric, Strideable, Hashable
...
+  ///   (In Swift 5.1 and earlier, isCanonical returns `true` for these
+  ///   values. This is a bug.)
+  ///
+  /// - On i386 and x86_64, `Float80` has a number of non-canonical
+  ///   encodings. "Pseudo-NaNs", "pseudo-infinities", and "unnormals" are
+  ///   interpreted as non-canonical NaN encodings. "Pseudo-denormals" are
+  ///   interpreted as non-canonical encodings of subnormal values.
+  ///
+  /// - Decimal floating-point admits a large number of non-canonical
+  ///   encodings. Consult the IEEE 754 standard for additional details.
```
[the remark this answers is not in the document]
Comment (stephentyrone) [reply stdlib/public/core/FloatingPoint.swift]: Will do.

**Comment by natecook1000** [comment]
Interesting, I hadn’t noticed that!

**Comment by stephentyrone** [comment]
@swift-ci please smoke test

**Review by natecook1000 — APPROVED** [review]

**Pull request merged**